services/motion/builtin/move_request.go

```diff
@@ -279,50 +278,154 @@ func (ms *builtIn) newMoveOnGlobeRequest(
+	if err != nil {
+		return nil, err
+	}
+
+	kb, err := kinematicbase.WrapWithKinematics(ctx, b, ms.logger, motion.NewSLAMLocalizer(slamSvc, true), limits, kinematicsOptions)
+	if err != nil {
+		return nil, err
+	}
+	goalPoseAdj := spatialmath.Compose(goalPoseRaw, motion.SLAMOrientationAdjustment)
+
+	// get point cloud data in the form of bytes from pcd
+	pointCloudData, err := slam.PointCloudMapFull(ctx, slamSvc)
+	if err != nil {
+		return nil, err
+	}
+	// store slam point cloud data  in the form of a recursive octree for collision checking
+	octree, err := pointcloud.ReadPCDToBasicOctree(bytes.NewReader(pointCloudData))
+	if err != nil {
+		return nil, err
+	}
+
+	mr, err := ms.relativeMoveRequestFromAbsolute(
+		ctx,
+		motionCfg,
+		ms.logger,
+		kb,
+		goalPoseAdj,
+		fs,
+		[]spatialmath.Geometry{octree},
+		valExtra,
+	)
+	return mr, err
+}
+
+func (ms *builtIn) relativeMoveRequestFromAbsolute(
+	ctx context.Context,
+	motionCfg *validatedMotionConfiguration,
+	logger logging.Logger,
+	kb kinematicbase.KinematicBase,
+	goalPoseInWorld spatialmath.Pose,
+	fs referenceframe.FrameSystem,
+	worldObstacles []spatialmath.Geometry,
+	valExtra validatedExtra,
+) (*moveRequest, error) {
 	// replace original base frame with one that knows how to move itself and allow planning for
 	kinematicFrame := kb.Kinematics()
-	if err = fs.ReplaceFrame(kinematicFrame); err != nil {
+	if err := fs.ReplaceFrame(kinematicFrame); err != nil {
 		return nil, err
 	}
 	// We want to disregard anything in the FS whose eventual parent is not the base, because we don't know where it is.
 	baseOnlyFS, err := fs.FrameSystemSubset(kinematicFrame)
 	if err != nil {
 		return nil, err
 	}
+
 	startPose, err := kb.CurrentPosition(ctx)
 	if err != nil {
 		return nil, err
 	}
-	startPoseToWorld := spatialmath.PoseInverse(startPose.Pose())
+	startPoseInv := spatialmath.PoseInverse(startPose.Pose())
 
-	goal := referenceframe.NewPoseInFrame(referenceframe.World, spatialmath.PoseBetween(startPose.Pose(), goalPoseRaw))
+	goal := referenceframe.NewPoseInFrame(referenceframe.World, spatialmath.PoseBetween(startPose.Pose(), goalPoseInWorld))
 
 	// convert GeoObstacles into GeometriesInFrame with respect to the base's starting point
-	geomsRaw := spatialmath.GeoObstaclesToGeometries(obstacles, origin)
-	geoms := make([]spatialmath.Geometry, 0, len(geomsRaw))
-	for _, geom := range geomsRaw {
-		geoms = append(geoms, geom.Transform(startPoseToWorld))
+	geoms := make([]spatialmath.Geometry, 0, len(worldObstacles))
+	for _, geom := range worldObstacles {
+		geoms = append(geoms, geom.Transform(startPoseInv))
 	}
 
 	gif := referenceframe.NewGeometriesInFrame(referenceframe.World, geoms)
 	worldState, err := referenceframe.NewWorldState([]*referenceframe.GeometriesInFrame{gif}, nil)
+	ms.logger.Infof("startPose: %v", spatialmath.PoseToProtobuf(startPose.Pose()))
+	ms.logger.Infof("requested world goal: %v", spatialmath.PoseToProtobuf(goalPoseInWorld))
```

Comment:
Maybe this is work for another ticket but we should be a little more regimented about how we log since we've been using the log files we create pretty extensively.  I'm thinking that we should have `String` methods for a lot of the structs we make and want to print/log.  So these could live on the moveRequest String method in the future, along with the rest of the info we want out of that struct

Reply:
IMO that should be a full, separate ticket and should touch nearly every single struct. If we do it top-down in one chunk like that, the formatting will be consistent.